### PR TITLE
refactor: rename LSP23 to LinkedContractsFactory

### PR DESCRIPTION
**Pull request by skimaharvey**
# What does this PR introduce? 

- rename LSP23 to LinkedContractsFactory
- rename controllerContract to primaryContract
- rename controlledContractOwner to secondaryContract
- fix tests

### CODE REVIEW

File: contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+    function _generateprimaryContractSalt(
+        PrimaryContractDeployment calldata primaryContractDeployment,
+        SecondaryContractDeployment calldata secondaryContractDeployment,
+        address postDeploymentModule,
+        bytes calldata postDeploymentModuleCalldata
+    ) internal pure virtual returns (bytes32 primaryContractGeneratedSalt) {
+        /* generate salt for the primary contract
+         *  the salt is generated by hashing the following elements:
+         *   - the salt
+         *   - the secondary contract bytecode
```
Comment (b00ste) [inline contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol:329]:
```suggestion
         *   - the primary contract bytecode
         *   - the secondary contract bytecode
```

File: contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+        /* initialize the primary contract proxy */
+        (bool success, bytes memory returnedData) = secondaryContractAddress
+            .call{value: msg.value}(secondaryInitializationBytes);
+        if (!success) {
+            revert SecondaryContractProxyInitFailureError(returnedData);
+        }
+    }
+
+    function _generateprimaryContractSalt(
```
Comment (b00ste) [inline contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol:320]:
```suggestion
    function _generatePrimaryContractSalt(
```

File: contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+
+        /* deploy the primary contract */
+        primaryContractAddress = Create2.deploy(
+            primaryContractDeployment.fundingAmount,
+            primaryContractGeneratedSalt,
+            primaryContractDeployment.creationBytecode
+        );
+    }
+
+    function _deploysecondaryContract(
```
Comment (b00ste) [inline contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol:228]:
```suggestion
    function _deploySecondaryContract(
```

File: contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+            primaryContractGeneratedSalt
+        );
+
+        secondaryContractAddress = Clones.predictDeterministicAddress(
+            secondaryContractDeploymentInit.implementationContract,
+            keccak256(abi.encodePacked(primaryContractAddress))
+        );
+    }
+
+    function _deployprimaryContract(
```
Comment (b00ste) [inline contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol:207]:
```suggestion
    function _deployPrimaryContract(
```

File: contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+        bytes calldata postDeploymentModuleCalldata
+    )
+        public
+        view
+        returns (
+            address primaryContractAddress,
+            address secondaryContractAddress
+        )
+    {
+        bytes32 primaryContractGeneratedSalt = _generateprimaryContractSalt(
```
Comment (b00ste) [inline contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol:142]:
```suggestion
        bytes32 primaryContractGeneratedSalt = _generatePrimaryContractSalt(
```

File: contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+        }
+
+        primaryContractAddress = _deployprimaryContract(
+            primaryContractDeployment,
+            secondaryContractDeployment,
+            postDeploymentModule,
+            postDeploymentModuleCalldata
+        );
+
+        secondaryContractAddress = _deploysecondaryContract(
```
Comment (b00ste) [inline contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol:47]:
```suggestion
        secondaryContractAddress = _deploySecondaryContract(
```

File: contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+        /* check that the msg.value is equal to the sum of the values of the primary and secondary contracts */
+        if (
+            msg.value !=
+            primaryContractDeployment.fundingAmount +
+                secondaryContractDeployment.fundingAmount
+        ) {
+            revert InvalidValueSum();
+        }
+
+        primaryContractAddress = _deployprimaryContract(
```
Comment (b00ste) [inline contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol:40]:
```suggestion
        primaryContractAddress = _deployPrimaryContract(
```

**Review by b00ste — APPROVED** [review]
Small typos, besides that LGTM 👍

**Review by CJ42 — APPROVED** [review]
Same suggestion than @b00ste

File: contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol (hunk 1 above)
Comment (skimaharvey) [reply contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol:329]: no here it is the secondary contract bytecode that need here

Comment (b00ste) [reply contracts/LSP23MultiChainDeployment/LinkedContractsFactory.sol:329]: its both

**Pull request merged**